In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

# Configure plots (correct way)
plt.style.use("seaborn-v0_8")   # <-- fix for matplotlib >= 3.6
sns.set_theme(style="whitegrid")

print("Setup complete. Libraries imported successfully!")


Setup complete. Libraries imported successfully!


## Loading DataSets and creating databases

### property DataFrame.shape[source]
Return a tuple representing the dimensionality of the DataFrame
### pandas.DataFrame.head
DataFrame.head(n=5)[source]
Return the first n rows.

This function returns the first n rows for the object based on position. It is useful for quickly testing if your object has the right type of data in it.

For negative values of n, this function returns all rows except the last |n| rows, equivalent to df[:n].

If n is larger than the number of rows, this function returns all rows.

In [ ]:
# Load dataset
train_path = "../data/application_train.csv"
test_path = "../data/application_test.csv"

df_train = pd.read_csv(train_path)
df_test = pd.read_csv(test_path)

# Check shapes
print("Train shape:", df_train.shape)
print("Test shape:", df_test.shape)

# Look at first few rows
df_train.head()


Train shape: (307511, 122)
Test shape: (48744, 121)


SK_ID_CURR  TARGET NAME_CONTRACT_TYPE CODE_GENDER FLAG_OWN_CAR  \
0      100002       1         Cash loans           M            N   
1      100003       0         Cash loans           F            N   
2      100004       0    Revolving loans           M            Y   
3      100006       0         Cash loans           F            N   
4      100007       0         Cash loans           M            N   

  FLAG_OWN_REALTY  CNT_CHILDREN  AMT_INCOME_TOTAL  AMT_CREDIT  AMT_ANNUITY  \
0               Y             0          202500.0    406597.5      24700.5   
1               N             0          270000.0   1293502.5      35698.5   
2               Y             0           67500.0    135000.0       6750.0   
3               Y             0          135000.0    312682.5      29686.5   
4               Y             0          121500.0    513000.0      21865.5   

   ...  FLAG_DOCUMENT_18 FLAG_DOCUMENT_19 FLAG_DOCUMENT_20 FLAG_DOCUMENT_21  \
0  ...                 0                0                0                0   
1  ...                 0                0                0                0   
2  ...                 0                0                0                0   
3  ...                 0                0                0                0   
4  ...                 0                0                0                0   

  AMT_REQ_CREDIT_BUREAU_HOUR AMT_REQ_CREDIT_BUREAU_DAY  \
0                        0.0                       0.0   
1                        0.0                       0.0   
2                        0.0                       0.0   
3                        NaN                       NaN   
4                        0.0                       0.0   

   AMT_REQ_CREDIT_BUREAU_WEEK  AMT_REQ_CREDIT_BUREAU_MON  \
0                         0.0                        0.0   
1                         0.0                        0.0   
2                         0.0                        0.0   
3                         NaN                        NaN   
4                         0.0                        0.0   

   AMT_REQ_CREDIT_BUREAU_QRT  AMT_REQ_CREDIT_BUREAU_YEAR  
0                        0.0                         1.0  
1                        0.0                         0.0  
2                        0.0                         0.0  
3                        NaN                         NaN  
4                        0.0                         0.0  

[5 rows x 122 columns]

## Check Target Distribution

In [5]:
# TARGET column = default indicator (0 = repaid, 1 = defaulted)
df_train['TARGET'].value_counts(normalize=True)


TARGET
0    0.919271
1    0.080729
Name: proportion, dtype: float64

## pandas.DataFrame.value_counts
DataFrame.value_counts(subset=None, normalize=False, sort=True, ascending=False, dropna=True)[source]
Return a Series containing the frequency of each distinct row in the Dataframe.

### Parameters:
subsetlabel or list of labels, optional
Columns to use when counting unique combinations.

### normalizebool, default False
Return proportions rather than frequencies.

### sortbool, default True
Sort by frequencies when True. Sort by DataFrame column values when False.

### ascendingbool, default False
Sort in ascending order.

### dropnabool, default True
Don’t include counts of rows that contain NA values.
### TARGET
0    0.919271   # 91.9% repaid

1    0.080729   # 8.1% defaulted
